src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1399,66 +1399,17 @@ internal static Encoding GetEncoding(string path)
                 return ClrFacade.GetDefaultEncoding();
             }
 
-            byte[] initialBytes = new byte[100];
-            int bytesRead = 0;
-
             try
             {
-                using (FileStream stream = System.IO.File.OpenRead(path))
+                using (StreamReader sr = new StreamReader(path, true))
                 {
-                    using (BinaryReader reader = new BinaryReader(stream))
-                    {
-                        bytesRead = reader.Read(initialBytes, 0, 100);
-                    }
+                    return sr.CurrentEncoding;
                 }
             }
             catch (IOException)
             {
                 return ClrFacade.GetDefaultEncoding();
```

Comment:
We could return utf8NoBom - we discussed about the default previously.

Reply:
In the catch you mean? I've seen that `ClrFacade.GetDefaultEncoding()` defaults also to utf8NoBom and it's used in a few other places. So I thought it was a bit cleaner to use this method instead of the static. Or should we replace all the calls to `GetDefaultEncoding` by the static in this class?

Reply:
Done in the update, used utf8NoBom and removed the use of the GetDefaultEncoding method